kaolin/render/spc/raytrace.py

```diff
@@ -48,4 +50,43 @@ def unbatched_raytrace(octree, points, pyramid, exsum, origin, direction, level)
         direction.contiguous(),
         level)
 
+def mark_first_hit(nuggets):
+    r"""Mark the first hit in the nuggets. 
+
+    Returns:
+        first_hits (torch.BoolTensor):
+    """
+    return _C.render.spc.mark_first_hit(nuggets.contiguous())
+
+def ray_aabb(nuggets, points, ray_o, ray_d, level, info=None, info_idxes=None, mask=None, init=True):
+    r"""Ray AABB intersection with points. 
+
+    Raytrace will already get correspondences, but this will additionally compute distances.
+    Likely to be folded into raytrace in a future version.
+
+    Args:
+        nuggets (torch.IntTensor):
+        points (torch.ShortTensor): the points associated to the octree.
+        ray_o (torch.FloatTensor): ray origins.
+        ray_d (torch.FloatTensor): ray directions.
+        level (int): level of the SPC to trace.
+        info (torch.BoolTensor): First hits. Will compute if None.
+        info_idxes (torch.IntTensor): Packed indices of first hits. Will compute if None.
+        mask (torch.BoolTensor): Mask to determine if the ray is still alive. Default to false if None.
+        init (bool): First run? Enables slight optimizations.
+
+    Returns
+        distance (torch.FloatTensor): distance from ray origin to ray-aabb intersection.
+        pidx (torch.LongTensor): corresponding point index.
+        mask (torch.BoolTensor): rewritten mask.
```

Comment:
Can you add some shape informations to the input / outputs ?

Reply:
Done ish